frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSquashAction.java

```diff
@@ -51,34 +49,37 @@ public LambdaLogger log() {
   protected void onUpdate(AnActionEvent anActionEvent) {
     super.onUpdate(anActionEvent);
 
-    Presentation presentation = anActionEvent.getPresentation();
+    val presentation = anActionEvent.getPresentation();
 
-    val branchNameOption = getNameOfBranchUnderAction(anActionEvent);
-    val nonRootBranchOption = branchNameOption.flatMap(bn -> getManagedBranchByName(anActionEvent, bn))
-        .flatMap(b -> b.isNonRoot() ? Option.of(b.asNonRoot()) : Option.none());
-    val syncToParentStatus = nonRootBranchOption.map(b -> b.getSyncToParentStatus()).getOrNull();
-    val numberOfCommits = nonRootBranchOption.map(b -> b.getCommits().length()).getOrNull();
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val managedBranch = getManagedBranchByName(anActionEvent, branchName);
+    val nonRootBranch = managedBranch != null && managedBranch.isNonRoot()
+        ? managedBranch.asNonRoot()
+        : null;
+    val syncToParentStatus = nonRootBranch != null ? nonRootBranch.getSyncToParentStatus() : null;
 
-    val branchName = branchNameOption.getOrNull();
-    if (branchName != null && nonRootBranchOption.isEmpty()) {
+    if (branchName != null && nonRootBranch == null) {
       presentation.setDescription(
           getNonHtmlString("action.GitMachete.BaseSquashAction.branch-is-root").format(branchName));
       presentation.setEnabled(false);
 
-    } else if (branchName != null && numberOfCommits != null) {
+    } else if (branchName != null && nonRootBranch != null) { // nonRootBranch is not null here.
+      val numberOfCommits = nonRootBranch.getCommits().length();
+
+      val description = getNonHtmlString("action.GitMachete.BaseSquashAction.not-enough-commits") //format is a static method, so it should not be called on instance.
+          // Now what is the role of getNonHtmlString?
```

Comment:
Oh, thanks for bringing it up. I wanted to ask for your input on the logic of this part of the code. It seems that we are not using the `description` string anywhere. Is this the desired logic?

Reply:
It is used on line 74 - `presentation.setDescription(description);`

Reply:
`val description = getNonHtmlString("action.GitMachete.BaseSquashAction.not-enough-commits").format(branchName, numberOfCommits + "", numberOfCommits == 1 ? "" : "s");`

This is the whole code for `description`. It seems that since `format` is a static method, calling it on `getNonHtmlString("action.GitMachete.BaseSquashAction.not-enough-commits")` would not make a difference. Or am I missing something?

Reply:
It is a static method, but using lombok it can be used as an extension to type String, similar to what can be done by the compiler of Scala 3. [Project Lombok](https://projectlombok.org/features/experimental/ExtensionMethod)